### PR TITLE
binary search method is complete, passes all tests

### PR DESCRIPTION
**Pull request by rnmessick**
- [x ] Top-level README “Table of Contents” is updated
 - [ x] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
     - [ x] “Happy Path” - Expected outcome
     - [x ] Expected failure
     - [x ] Edge Case (if applicable/obvious)
 - [x ] README for this challenge is complete
     - [x ] Summary, Description, Approach & Efficiency, Solution
     - [ x] Link to code
     - [x ] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            if (sortedArray[midpoint] < key) {
+                low = midpoint + 1;
+            } else if (sortedArray[midpoint] > key) {
+                max = midpoint - 1;
+            } else if (sortedArray[midpoint] == key) {
+                return midpoint;
+            }
+        }
+        // Check that sortedArray is not empty and for the key to not be in it
+        if (sortedArray.length > 0 && sortedArray[low] == key) {
```
Comment (evanslaton) [inline code401challenges/src/main/java/code401challenges/BinarySearch.java]: You might consider putting checks for certain edge cases at the beginning of the method. If you check if the input array is empty at the start of the method then the rest of your code won't need to run (returning -1 still needs to be at the bottom)
